### DISCUSSION TITLE
How to add file to project to make it work in built image

### DISCUSSION THREAD

**Discussion by yury98** [apache/camel-karavan#1329] [Q&A] [answered]
### Ask a question

Hi!

I'm trying to make integration with JSONata component. I've created file mapping.json in project and set `resourceUri: file:mapping.json`. It works fine in devmode, but I get an exception `java.io.FileNotFoundException: mapping.json does not exist ` when I build image and start it. What shold I do to make it work?

### Variant

Web Application

### Container Management (if applicable)

Docker

### Operating System (if applicable)

None

### Version

4.6.0

**Comment by mgubaidullin** [comment]
could you try `resourceUri: mapping.json` (classpath by default)

**Reply by yury98** [reply]
It works in build, but not in devmode. I get just `java.io.FileNotFoundException: Cannot find resource: mapping.json for URI: mapping.json ` in devmode

**Reply by davsclaus** [reply]
Can you put your example source code somewhere where we can try and download

**Comment by filojalo** [comment, marked as answer]
I had a similar problem with a xsl file. When using `resourceUri: file:jsonToXml.xsl` it does not work in the built container.
However using the syntax `resourceUri: jsonToXml.xsl` worked for me both in devmode and built container.

Using Karavan and Camel 4.8.0.. so most likely fixed in newer versions and I think this thread can be closed.
